src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/RoleAssignmentsFilteringServiceImpl.java

```diff
@@ -28,62 +24,33 @@ public RoleAssignmentsFilteringServiceImpl(List<RoleAttributeMatcher> roleAttrib
     }
 
     @Override
-    public RoleAssignmentFilteringResult filter(RoleAssignments roleAssignments,
+    public List<RoleAssignment>  filter(RoleAssignments roleAssignments,
                                                 CaseDetails caseDetails) {
         log.info("Filter role assignments for case {}", caseDetails.getReference());
 
-        List<Pair<RoleAssignment, RoleMatchingResult>> roleAssignmentMatchPairs = roleAssignments
+        return roleAssignments
             .getRoleAssignments()
             .stream()
-            .map(roleAssignment -> {
-                Pair<RoleAssignment, RoleMatchingResult> resultPair = Pair.of(roleAssignment, new RoleMatchingResult());
-
-                roleAttributeMatchers.forEach(matcher -> matcher.matchAttribute(resultPair, caseDetails));
-                return resultPair;
-            })
-            .filter(resultPair -> resultPair.getRight().matchedAllValues())
-            .collect(Collectors.toList());
-
-        return new RoleAssignmentFilteringResult(roleAssignmentMatchPairs);
-    }
-
-    @Override
-    public RoleAssignmentFilteringResult filter(RoleAssignments roleAssignments,
-                                                      CaseDataContent caseDataContent) {
-        log.info("Filter role assignments for case event {}", caseDataContent.getEvent().getEventId());
-
-        List<Pair<RoleAssignment, RoleMatchingResult>> roleAssignmentMatchPairs = roleAssignments
-            .getRoleAssignments()
-            .stream()
-            .map(roleAssignment -> {
-                Pair<RoleAssignment, RoleMatchingResult> resultPair = Pair.of(roleAssignment, new RoleMatchingResult());
-
-                roleAttributeMatchers.forEach(matcher -> matcher.matchAttribute(resultPair, (CaseTypeDefinition) null));
-                return resultPair;
-            })
-            .filter(resultPair -> resultPair.getRight().matchedAllValues())
+            .filter(roleAssignment -> roleAttributeMatchers
+                .stream()
+                .map(matcher -> matcher.matchAttribute(roleAssignment, caseDetails))
+                .allMatch(matched -> matched == true))
```

Comment:
```suggestion
                .allMatch(matcher -> matcher.matchAttribute(roleAssignment, caseDetails)))
```
And same simplification for the other methods

Reply:
Could also consider a common method for some of the repeated code.. e.g. introduce a new method
```
    private List<RoleAssignment> filterMatchingRoleAssignments(RoleAssignments roleAssignments, BiPredicate<RoleAttributeMatcher, RoleAssignment> hasMatch) {
        return roleAssignments
            .getRoleAssignments()
            .stream()
            .filter(roleAssignment -> roleAttributeMatchers
                .stream()
                .allMatch(matcher -> hasMatch.test(matcher, roleAssignment)))
            .collect(Collectors.toList());
    }
``` 
which would turn this into:
```
return filterMatchingRoleAssignments(roleAssignments,
            (matcher, roleAssignment) -> matcher.matchAttribute(roleAssignment, caseDetails));
```